### PR TITLE
WL-3576 : Adding logic for email sender/recipient in Contact Us Tool

### PR DESCRIPTION
**Pull request by nicholaswilson100**

### CODE REVIEW

File: src/java/org/sakaiproject/feedback/tool/FeedbackTool.java [hunk 1]
```diff
@@ -32,7 +33,12 @@
 
     private SakaiProxy sakaiProxy = null;
 
-    public void init(ServletConfig config) throws ServletException {
+    private SecurityService securityService = null;
+
+    private SiteService siteService = null;
+
+
+	public void init(ServletConfig config) throws ServletException {
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/FeedbackTool.java]: Indentation?

File: src/java/org/sakaiproject/feedback/tool/FeedbackTool.java [hunk 2]
```diff
@@ -59,11 +68,17 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
-                // pick one
+            Site site;
+            try {
+                site = siteService.getSite(siteId);
+            } catch (IdUnusedException e) {
+                throw new RuntimeException("The site cannot be found with siteId: " + siteId, e);
+            }
+
+            Map<String, String> siteUpdaters = new HashMap<String, String>();
+            boolean hasViewPermission = securityService.unlock((userId!=null ? userId : ""), "roster.viewallmembers", site.getReference());
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/FeedbackTool.java]: How can userId be null? as you're checking it at line 69.

But I don't think you need to pass in the userId at all do you as it should just use the current one for the request.
`org.sakaiproject.authz.api.SecurityService#unlock(java.lang.String, java.lang.String)`

Comment (nicholaswilson100) [inline src/java/org/sakaiproject/feedback/tool/FeedbackTool.java]: Doh! yes you're right...

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java [hunk 3]
```diff
@@ -139,39 +129,11 @@ private String handleReport(final EntityView view, final Map<String, Object> par
...
-                    logger.error("No sender email address for non logged in user. Returning BAD REQUEST ...");
-                    return BAD_REQUEST;
-                }
-            } else {
-                senderAddress = sakaiProxy.getUser(userId).getEmail();
-            }
-        } else {
-            senderAddress = sakaiProxy.getUser(userId).getEmail();
+        if (userId != null) {
+			senderAddress = sakaiProxy.getUser(userId).getEmail();
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java]: Indentation.

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java [hunk 4]
```diff
@@ -189,9 +151,39 @@ private String handleReport(final EntityView view, final Map<String, Object> par
                 }
             } else {
                 // The site has a contact email. Use it as the toAddress.
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java]: The comment isn't correct.

File: src/java/org/sakaiproject/feedback/util/SakaiProxy.java [hunk 5]
```diff
@@ -259,10 +255,8 @@ public void sendEmail(String fromUserId, String senderAddress, String toAddress,
         	}
         }
 
-		if (feedbackType.equals(Constants.CONTENT)) {
-            // Copy the sender in
-			msg.addRecipient(RecipientType.CC, fromName, fromAddress);
-		}
+        // Copy the sender in
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/util/SakaiProxy.java]: Indentation.

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java [hunk 6]
```diff
@@ -189,9 +151,39 @@ private String handleReport(final EntityView view, final Map<String, Object> par
...
-                toAddress = sakaiProxy.getSiteProperty(siteId, Site.PROP_SITE_CONTACT_EMAIL);
+                toAddress = sakaiProxy.getConfigString(Constants.PROP_TECHNICAL_ADDRESS, null);
             }
-        }
+
+		}
+		else {
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java]: Code style change

**Comment by buckett** [comment]
FeedbackEntityProvider.java#141 Still references the old logic.

File: src/java/org/sakaiproject/feedback/tool/FeedbackTool.java [hunk 7]
```diff
@@ -59,11 +68,17 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 
         if (userId != null) {
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/FeedbackTool.java]: According to the logic in the Jira we weren't going todo checks to see if the user was logged in.

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java [hunk 8]
```diff
@@ -189,9 +151,39 @@ private String handleReport(final EntityView view, final Map<String, Object> par
                 }
             } else {
                 // The site has a contact email. Use it as the toAddress.
-                toAddress = sakaiProxy.getSiteProperty(siteId, Site.PROP_SITE_CONTACT_EMAIL);
+                toAddress = sakaiProxy.getConfigString(Constants.PROP_TECHNICAL_ADDRESS, null);
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java]: If the user is logged in and doesn't have permission to see the site members then the email will always go to the central technical address and never the site contact email.

File: src/java/org/sakaiproject/feedback/util/SakaiProxy.java (hunk 5 above)
Comment (nicholaswilson100) [inline src/java/org/sakaiproject/feedback/util/SakaiProxy.java]: Fixed below.

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java (hunk 6 above)
Comment (nicholaswilson100) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java]: Got it

File: src/java/org/sakaiproject/feedback/tool/FeedbackTool.java (hunk 7 above)
Comment (nicholaswilson100) [inline src/java/org/sakaiproject/feedback/tool/FeedbackTool.java]: I've moved the permission stuff out of the userId==null check.

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java (hunk 4 above)
Comment (nicholaswilson100) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java]: Removed comment

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java (hunk 8 above)
Comment (nicholaswilson100) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java]: Good spot. Added logic to say that the email for a logged in user with no permission is the site contact email or the technical one if that doesn't exist.

**Comment by buckett** [comment]
Won't merge cleanly.

**Comment by nicholaswilson100** [comment]
Hmmm... should we merge it un-cleanly then I can fix it up after?

**Pull request merged**